### PR TITLE
[Add Keyboard] Add the Compensator

### PR DESCRIPTION
**Pull request by Lrfoster03**
Added support for the Compensator Keyboard, seen [here](https://geekhack.org/index.php?topic=122493.0)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/compensator/matrix_diagram.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Matrix Diagram for Compensator
+
+```
+┌───────┐┌───────────────┐┌───────────┐┌───────────────────────────────────────────────────────────┐┌───────────┐┌───────────────┐┌───────┐
+│00 │01 ││02 │03 │04 │05 ││06 │07 │08 ││09   │0A │0B │0C │0D │0E │0F │0G │0H │0I │0J │0K │0L │0M   ││0N │0O │0P ││0Q │0R │0S │0T ││0U │0V │
+├───┼───┤├───┼───┼───┼───┤├───┼───┼───│├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤├───┼───┼───│├───┼───┼───┼───┤├───┼───┤
+│10 │11 ││12 │13 │14 │15 ││16 │17 │18 ││19    │1A │1B │1C │1D │1E │1F │1G │1H │1I │1J │1K │1L      ││1M │1N │1O ││1P │1Q │1R │1S ││1T │1U │
+├───┼───┤├───┼───┼───┼───┤└───┼───┼───┘├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┤└───┼───┼───┘├───┼───┼───┼───┤├───┼───┤
+│20 │21 ││22 │23 │24 │25 │    │26 │    │27  │28 │29 │2A │2B │2C │2D │2E │2F │2G │2H │2I │2J    │2K │    │1L │    │2M │2N │2O │2P ││2Q │2R │
```
Comment (fauxpark) [inline keyboards/compensator/matrix_diagram.md:9]:
```suggestion
│20 │21 ││22 │23 │24 │25 │    │26 │    │27  │28 │29 │2A │2B │2C │2D │2E │2F │2G │2H │2I │2J    │2K │    │1L │    │2M │2N │2O │2P ││2Q │2R │
```

File: keyboards/compensator/matrix_diagram.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+│20 │21 ││22 │23 │24 │25 │    │26 │    │27  │28 │29 │2A │2B │2C │2D │2E │2F │2G │2H │2I │2J    │2K │    │1L │    │2M │2N │2O │2P ││2Q │2R │      
+├───┼───┤├───┼───┼───┼───┤┌───┼───┼───┐├────┴┬──┴┬──┴──┬┴───┴───┴───┴───┴───┴───┴──┬┴───┴┬───┬─┴───┤┌───┼───┼───┐├───┼───┼───┼───┤├───┼───┤
+│31 │32 ││33 │34 │35 │36 ││37 │38 │39 ││3A   │   │3B   │3C                         │3D   │   │3E   ││3F │3G │3H ││3I │3J │3K │3L ││30 │31 │
+└───┴───┘└───┴───┴───┴───┘└───┴───┴───┘└─────┘   └─────┴───────────────────────────┴─────┘   └─────┘└───┴───┴───┘└───┴───┴───┴───┘└───┴───┘
+                 ┌───────┐             ┌────────┐                                       ┌──────────┐             ┌───────┐
+  2u Numpad Zero │36     │             │27      │ 2.25u LShift             2.75u RShift │2J        │             │3I     │ 2u Numpad Zero
+                 └───────┘             └────────┘                                       └──────────┘             └───────┘
+         ┌───┐                                    ┌───────────────────────────────────────┐                                  ┌───┐
```
Comment (fauxpark) [inline keyboards/compensator/matrix_diagram.md:16]:
```suggestion
         ┌───┐                                    ┌───────────────────────────────────────┐                                  ┌───┐
```

File: keyboards/compensator/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,                          KC_TRNS, KC_TRNS,  KC_TRNS,         KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                       KC_TRNS,  KC_TRNS, KC_TRNS,         KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,                                                                                                        KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS
+    )
```
Comment (fauxpark) [inline keyboards/compensator/keymaps/via/keymap.c:42]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/compensator/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+enum custom_keycodes {
+    KC_P00 = QK_KB_0
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base -> XT:Num:Nav:Alphas:Nav:Num:Xt*/
+        KC_F1,  KC_F2,     KC_PPLS,  KC_P7, KC_P8, KC_P9,          KC_INS,  KC_HOME, KC_PGUP,          KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,                KC_INS,  KC_HOME, KC_PGUP,          KC_P7,   KC_P8,   KC_P9,   KC_PPLS,     KC_F1,  KC_F2,
+        KC_F3,  KC_F4,     KC_PMNS,  KC_P4, KC_P5, KC_P6,          KC_DEL,  KC_END,  KC_PGDN,          KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                          KC_DEL,  KC_END,  KC_PGDN,          KC_P4,   KC_P5,   KC_P6,   KC_PMNS,     KC_F3,  KC_F4,
+        KC_F5,  KC_F6,     KC_PENT,  KC_P1, KC_P2, KC_P3,          KC_NO,   KC_UP,   KC_NO,            KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                  KC_NO,   KC_UP,   KC_NO,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,     KC_F5,  KC_F6,
```
Comment (zvecr) [inline keyboards/compensator/keymaps/default/keymap.c:26]: `MO(1)` would go out of bounds

File: keyboards/compensator/info.json [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": true,
+        "command": true,
+        "nkro": false,
+        "backlight": false,
+        "rgblight": false,
+        "audio": false,
+        "key_lock": false
```
Comment (zvecr) [inline keyboards/compensator/info.json:28]:
```suggestion
        "nkro": false
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/compensator/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+    KC_P00 = QK_KB_0
```
Comment (drashna) [inline keyboards/compensator/keymaps/default/keymap.c:19]:
```suggestion
    KC_P00 = SAFE_RANGE,
```
or
```suggestion
    KC_P00 = QK_USER,
```

File: keyboards/compensator/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+    KC_P00 = QK_KB_0
```
Comment (drashna) [inline keyboards/compensator/keymaps/via/keymap.c:19]: same.

**Review by drashna — COMMENTED** [review]
That or move the macro to compensator.c/h

File: keyboards/compensator/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,                          KC_TRNS, KC_TRNS,  KC_TRNS,         KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                       KC_TRNS,  KC_TRNS, KC_TRNS,         KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,                                                                                                        KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,     KC_TRNS,  KC_TRNS
```
Comment (waffle87) [inline keyboards/compensator/keymaps/default/keymap.c:34]:
```suggestion
        KC_F5,  KC_F6,     KC_PENT,  KC_P1, KC_P2, KC_P3,          KC_NO,   KC_UP,   KC_NO,            KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_TRNS,                  KC_NO,   KC_UP,   KC_NO,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,     KC_F5,  KC_F6,
        KC_F7,  KC_F8,     KC_PMNS,  KC_PDOT, KC_P0, KC_P00,       KC_LEFT, KC_DOWN, KC_RIGHT,         KC_LCTL, KC_LALT, KC_SPC,  KC_SPC,  KC_SPC,  KC_LALT, KC_LCTL,                                                                               KC_LEFT, KC_DOWN, KC_RIGHT,         KC_P00,  KC_P0,  KC_PDOT,  KC_PMNS,     KC_F7,  KC_F8
```

File: keyboards/compensator/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+enum custom_keycodes {
+    KC_P00 = SAFE_RANGE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base -> XT:Num:Nav:Alphas:Nav:Num:Xt*/
+        KC_F1,  KC_F2,     KC_PPLS,  KC_P7, KC_P8, KC_P9,          KC_INS,  KC_HOME, KC_PGUP,          KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,                KC_INS,  KC_HOME, KC_PGUP,          KC_P7,   KC_P8,   KC_P9,   KC_PPLS,     KC_F1,  KC_F2,
+        KC_F3,  KC_F4,     KC_PMNS,  KC_P4, KC_P5, KC_P6,          KC_DEL,  KC_END,  KC_PGDN,          KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                          KC_DEL,  KC_END,  KC_PGDN,          KC_P4,   KC_P5,   KC_P6,   KC_PMNS,     KC_F3,  KC_F4,
+        KC_F5,  KC_F6,     KC_PENT,  KC_P1, KC_P2, KC_P3,          KC_NO,   KC_UP,   KC_NO,            KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                  KC_NO,   KC_UP,   KC_NO,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,     KC_F5,  KC_F6,
```
Comment (waffle87) [inline keyboards/compensator/keymaps/via/keymap.c:26]:
```suggestion
        KC_F5,  KC_F6,     KC_PENT,  KC_P1, KC_P2, KC_P3,          KC_NO,   KC_UP,   KC_NO,            KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_TRNS,                  KC_NO,   KC_UP,   KC_NO,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,     KC_F5,  KC_F6,
```

**Comment by Lrfoster03** [comment]
Initially, I wanted to leave the layers for the user to customize themselves, but I should've mentioned this. I believe this change will be better for the end-user anyway.

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**